pkg/kube/activity.go

```diff
@@ -200,6 +202,16 @@ func (k *PipelineActivityKey) GetOrCreate(jxClient versioned.Interface, ns strin
 		create = true
 		a = defaultActivity
 	}
+
+	numberOfRefs := len(k.PullRefs)
```

Comment:
I'd make this a function of `PipelineActivityKey`

Reply:
Agree, created two new funcs: isBatchBuild and isPRBuild. Should make the code much cleaner.